thorlcr/activities/indexread/thindexread.cpp

```diff
@@ -179,7 +182,8 @@ class CIndexReadBase : public CMasterActivity
                         rfn.getPath(remotePath);
                         unsigned crc = 0;
                         lastPart->getCrc(crc);
-                        keyIndex.setown(createKeyIndex(remotePath.str(), crc, false, 0));
+                        size32_t blockedIOSize = getEffectiveBlockSize();
+                        keyIndex.setown(createKeyIndex(remotePath.str(), crc, false, blockedIOSize));
```

Comment:
NB: This, thindexread.cpp, and all other activity files not ending with *slave.cpp are the Thor manager implementations, they do minimal work/minimal reading of data (if any).
In this case, this manager side of indexreadbase is opening up the TLK, a small top-level-key that has the 1st row of each part in it. So the blocksize change is of minimal help - still correct to do so though, but it's the slave/worker code that reads the row data itself and needs to use blockSize to do so efficiently.